content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java

```diff
@@ -1003,26 +958,149 @@ private Date getDueDateRetryTime(Date dueDate) {
+			return false;
+		}
+	}
+
+	private boolean checkForDraftFlag(ContentReviewItem item) {
+		// Checks if current item is a draft or submitted by it's review score, does not require access to assignmentService
+		try {
+			return DRAFT_ITEM_REVIEW_SCORE.equals(item.getReviewScore());
+		} catch (Exception e) {
+			log.error(e.getMessage(), e);
+			// Error retrieving draft, process item as final submission
+			return false;
+		}
+	}
+
 	private void createPlaceholderItem(ContentReviewItem item, Date dueDate) {
 		log.info("Creating placeholder item for when due date is passed for ItemID: " + item.getId());						
 		ContentReviewItem placeholderItem = new ContentReviewItem();
 		// Review score is used as flag for placeholder items in checkForReport
 		placeholderItem.setReviewScore(PLACEHOLDER_ITEM_REVIEW_SCORE); 
 		// Content Id must be original
-		placeholderItem.setContentId(item.getContentId() + PLACEHOLDER_STRING_FLAG);	
+		placeholderItem.setContentId(item.getContentId() + PLACEHOLDER_STRING_FLAG);
+		// This is needed for webhook call, without it external id query does not return a single item
+		placeholderItem.setExternalId(item.getExternalId() + PLACEHOLDER_STRING_FLAG);
 		placeholderItem.setStatus(ContentReviewConstants.CONTENT_REVIEW_SUBMITTED_AWAITING_REPORT_CODE);
 		placeholderItem.setNextRetryTime(getDueDateRetryTime(dueDate));
 		placeholderItem.setDateQueued(new Date());
 		placeholderItem.setDateSubmitted(new Date());
 		placeholderItem.setRetryCount(new Long(0));	
-		// All other fields are copied from original item 
-		placeholderItem.setExternalId(item.getExternalId());
+		// All other fields are copied from original item
 		placeholderItem.setProviderId(item.getProviderId());						
 		placeholderItem.setUserId(item.getUserId());
 		placeholderItem.setSiteId(item.getSiteId());
 		placeholderItem.setTaskId(item.getTaskId());																	
 		crqs.update(placeholderItem);
-	}			
+	}
+
+	private void handleSubmissionStatus(String submissionStatus, ContentReviewItem item, Assignment assignment) {
+		try {
+
+			Date assignmentDueDate = Date.from(assignment.getDueDate());
+			String reportGenSpeed = assignment.getProperties().get("report_gen_speed");
+
+			// Handle possible error status
+			String errorStr = null;
+
+			switch (submissionStatus) {
+			case "COMPLETE":
+				// Check for items that generate reports both immediately and on due date or draft items
+				// Create a placeholder item that will regenerate and index report after due date
+				if (assignmentDueDate != null && assignmentDueDate.after(new Date())
+						&& GENERATE_REPORTS_IMMEDIATELY_AND_ON_DUE_DATE.equals(reportGenSpeed) || DRAFT_ITEM_REVIEW_SCORE.equals(item.getReviewScore())) {
+					createPlaceholderItem(item, assignmentDueDate);
+				}
+				// If submission status is complete, start similarity report process
+				generateSimilarityReport(item.getExternalId(), item.getTaskId(), checkForDraftFlag(item));
```

Comment:
Use the assignment submission "isDraft" flag instead of storing the draft status as DRAFT_ITEM_REVIEW_SCORE in the item score. If this is done, you will need to use a SecurityAdvisor to override permissions for the webhook user (there may also be a way to do this with the session manager)